extensions/azurecore/src/azureResource/services/terminalService.ts

```diff
@@ -0,0 +1,185 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as nls from 'vscode-nls';
+import * as request from 'request-promise';
+import * as WS from 'ws';
+
+import { IAzureTerminalService } from '../interfaces';
+import { AzureAccount, AzureAccountSecurityToken, Tenant } from '../../account-provider/interfaces';
+
+const localize = nls.loadMessageBundle();
+export class AzureTerminalService implements IAzureTerminalService {
+	private readonly apiVersion = '?api-version=2018-10-01';
+
+	public constructor(context: vscode.ExtensionContext) {
+
+	}
+
+	public async getOrCreateCloudConsole(account: AzureAccount, tenant: Tenant, tokens: { [key: string]: AzureAccountSecurityToken }): Promise<void> {
+		const metadata = account.properties.providerSettings;
+		const consoleRequestUri = this.getConsoleUri(metadata.settings.armResource.endpoint);
+		const token = tokens[tenant.id].token;
+		const provisionRequest = request({
+			uri: consoleRequestUri,
+			method: 'PUT',
+			headers: {
+				'Accept': 'application/json',
+				'Content-Type': 'application/json',
+				'Authorization': `Bearer ${token}`
+			},
+			json: true,
+			body: {
+				properties: {
+					'osType': 'linux'
+				}
+			}
+		});
+
+		const provisionResult = await provisionRequest;
+
+		if (provisionResult?.properties?.provisioningState !== 'Succeeded') {
+			throw new Error(provisionResult);
+		}
+
+		const consoleUri = provisionResult.properties.uri;
+		return this.createTerminal(consoleUri, token, account.displayInfo.displayName);
+	}
+
+
+	private async createTerminal(provisionedUri: string, token: string, accountDisplayName: string): Promise<void> {
+		let shell = await vscode.window.showQuickPick(['Bash', 'PowerShell'], { canPickMany: false });
```

Comment:
Because I haven't done this in basically most of the Azure codebase I'll make a seperate PR in the future when I want to do this kind of testing and fix all the other problems.